Source/FullScreenPokemon.ts

```diff
@@ -4829,6 +4829,64 @@ module FullScreenPokemon {
                 callback);
         }
 
+        /* partyActivate functions
+         */
+
+        /**
+         * Makes sure the Player has a Pokemon in the party that knows the necessary HM move.
+         * 
+         * @param player   The Player.
+         * @param thing   The Solid to be affected.
+         * @todo Eventually add check to make sure the Player beat the Gym leader needed to use the move.
+         */
+        checkPartyActivate(player: IPlayer, thing: IHMSolid): void {
+            var partyPokemon: IPokemon[] = player.FSP.ItemsHolder.getItem("PokemonInParty"),
+                moves: BattleMovr.IMove[],
+                move: IMoveSchema,
+                i: number,
+                j: number;
+
+            for (i = 0; i < partyPokemon.length; i += 1) {
+                moves = partyPokemon[i].moves;
+
+                for (j = 0; j < moves.length; j += 1) {
+                    if (moves[j].title === thing.moveName) {
+                        move = player.FSP.MathDecider.getConstant("moves")[moves[j].title];
+                        player.FSP.activateMove(move, player, partyPokemon[i]);
+                    }
+                }
+            }
```

Comment:
Since using the math constants is becoming a pain, lets move the move callback to objects.js. So, you'd have `moveName` and `moveCallback` (or similarly named).